beacon/blockchain/payload.go

```diff
@@ -52,6 +56,74 @@ func (s *Service) forceStartupHead(
 	}
 }
 
+// forceSyncUponFinalize sends a new payload and force startup FCU to the Execution
+// Layer client. This informs the EL client of the new head and forces a SYNC
+// if blocks are missing. This function should only be run once at startup.
+func (s *Service) forceSyncUponFinalize(
+	ctx context.Context,
+	beaconBlock *ctypes.BeaconBlock,
+) error {
+	// NewPayload call first to load payload into EL client.
+	executionPayload := beaconBlock.GetBody().GetExecutionPayload()
+	parentBeaconBlockRoot := beaconBlock.GetParentBlockRoot()
+	payloadReq := ctypes.BuildNewPayloadRequest(
+		executionPayload,
+		beaconBlock.GetBody().GetBlobKzgCommitments().ToVersionedHashes(),
+		&parentBeaconBlockRoot,
+	)
+	if err := payloadReq.HasValidVersionedAndBlockHashes(); err != nil {
```

Comment:
I was the main proposer of pulling `HasValidVersionedAndBlockHashes` out of executionEngine and just have `NotifyNewPayload`, but now it's annoying to have to call `HasValidVersionedAndBlockHashes` explicitly.
We should consider letting `BuildNewPayloadRequest` return an error and have `HasValidVersionedAndBlockHashes` called in it.